client/state/ui/notices/actions.js

```diff
@@ -0,0 +1,49 @@
+/**
+ * Internal dependencies
+ */
+import {
+	NEW_NOTICE,
+	REMOVE_NOTICE
+} from 'state/action-types';
+
+import { uniqueId } from 'lodash';
+
+export function createNoticeAction( status, text, options = {} ) {
+	return {
+		noticeId: uniqueId(),
+		duration: options.duration,
+		showDismiss: ( typeof options.showDismiss === 'boolean' ? options.showDismiss : true ),
```

Comment:
Rather than type-checking, if the goal is to default to `true` if not defined, we could check for the presence of the key:

```jsx
showDismiss: 'showDismiss' in options ? options.showDismiss : true
```